packages/vscode-extension/src/project/project.ts

```diff
@@ -597,6 +597,9 @@ export class Project
 
       const previewURL = await newDeviceSession.start(this.deviceSettings, {
         cleanBuild: forceCleanBuild,
+        previewReadyCallback: (url) => {
+          this.updateProjectStateForDevice(deviceInfo, { previewURL: url });
+        },
       });
       this.updateProjectStateForDevice(this.projectState.selectedDevice!, {
```

Comment:
This is called twice with `previewURL`, some explanation why is that could be useful here (I assume that ideally, the second call would just set the status to `"running"`, since the `previewURL` has been already set at this point but I don't know the codebase).

Reply:
+1, couldn't the second one update only "status"? then the `previewURL` local variable is redundant too